core/src/jsMain/kotlin/io/islandtime/zone/TimeZoneRules.kt

```diff
@@ -0,0 +1,313 @@
+@file:Suppress("NewApi")
+
+package io.islandtime.zone
+
+
+import io.islandtime.*
+import io.islandtime.intl.DateTimeFormat
+import io.islandtime.measures.*
+import kotlin.js.Date
+
+actual object PlatformTimeZoneRulesProvider : TimeZoneRulesProvider {
+
+    override val databaseVersion: String
+        get() = "1"
+
+    override val availableRegionIds: Set<String>
+        get() = setOf()
+
+    override fun hasRulesFor(regionId: String): Boolean {
+        return try {
+            DateTimeFormat("en-US", objectOf {
+                timeZone = regionId
+            }).format()
+            true
+        } catch (e: Exception) {
+            false
+        }
+    }
+
+    override fun rulesFor(regionId: String): TimeZoneRules {
+        val lowered = regionId.toLowerCase()
+        return when {
+            lowered === "local" -> LocalRules()
```

Comment:
The reason for "local" is because `DateTimeFormat` might return "local" as the timezone from the system.
I saw this behavior in Luxon.
I'm not really sure how we can remove "local"

Reply:
I don't think "local" is ever returned by the system -- it's just something Luxon uses internally. The zone that's returned in `resolvedOptions` should be an IANA region ID.